lib/oli_web/router.ex

```diff
@@ -835,6 +835,12 @@ defmodule OliWeb.Router do
         :preview
       )
     end
+
+    get(
+      "/downloads/progress/:container_id",
+      MetricsController,
+      :download_container_progress
+    )
```

Comment:
Since this is downloading a file, I don't think this is the best place to put this route. Maybe we can have it under the `/api` scope like some other endpoints we have?

```suggestion
    scope "/api/v1/sections/:section_slug"
      pipe_through([
        :browser,
        :delivery_and_admin,
        :maybe_gated_resource,
        :pow_email_layout
      ])
      
      get(
        "/:container_id/progress",
        MetricsController,
        :download_container_progress
      )
  end
```

Reply:
I don't think `/api` is the right place for downloading file routes, only since these routes all originate from direct user interaction (e.g. a button or a link in the UX is clicked to trigger it) as opposed to a piece of a client side typescript code invoking the endpoint.

Reply:
Yes, you're right that we're invoking `/api` endpoints just from the client side. I still think it's a little bit weird that it's under the `/sections/:section_slug/instructor_dashboard` scope but maybe if we use it somewhere else in the future, we can move it then